File: source/bindings/csharp/bot_connector_config.cs
//
// Copyright (c) Microsoft. All rights reserved.
// See https://aka.ms/csspeech/license201809 for the full license information.
//
using System;
using Microsoft.CognitiveServices.Speech.Internal;
using static Microsoft.CognitiveServices.Speech.Internal.SpxExceptionThrower;

namespace Microsoft.CognitiveServices.Speech.Dialog
{
    /// <summary>
    /// Class that defines configurations for speech bot connector
    /// Added in version 1.5.0
    /// </summary>
    public sealed class BotConnectorConfig : SpeechConfig
    {
        internal BotConnectorConfig(IntPtr handle) : base(handle)
        {
        }

        /// <summary>
        /// Creates an instance of the bot connector config with the specified Speech Channel secret key.
        /// </summary>
        /// <param name="secretKey">Speech Channel secret key.</param>
        /// <param name="subscription">Subscription key associated with the bot</param>
        /// <param name="region">The region name (see the <a href="https://aka.ms/csspeech/region">region page</a>).</param>
        /// <returns>A new bot connector config.</returns>
        public static BotConnectorConfig FromSecretKey(string secretKey, string subscription, string region)
        {
            IntPtr botConfigHandle = IntPtr.Zero;
            ThrowIfFail(Internal.BotConnectorConfig.bot_connector_config_from_secret_key(out botConfigHandle, secretKey, subscription, region));
            return new BotConnectorConfig(botConfigHandle);
        }

        /// <summary>
        /// Speech Channel secret key.
        /// </summary>
        public string SecretKey
        {
            get
            {
                return progBag.GetProperty(PropertyId.Conversation_Secret_Key);
            }
            set
            {
                progBag.SetProperty(PropertyId.Conversation_Secret_Key, value);
            }
        }

        /// <summary>
        /// Bot Connection initial silence timeout property.
        /// </summary>
        public string BotInitialSilenceTimeout
        {
            get
            {
                return progBag.GetProperty(PropertyId.Conversation_Initial_Silence_Timeout);
            }
            set
            {
                progBag.SetProperty(PropertyId.Conversation_Initial_Silence_Timeout, value);
            }
        }

        /// <summary>
        /// The format of the audio that is returned during text to speech.
        /// </summary>
        public string TextToSpeechAudioFormat
        {
            get
            {
                return progBag.GetProperty(PropertyId.SpeechServiceConnection_SynthOutputFormat);
            }
            set
            {
                progBag.SetProperty(PropertyId.SpeechServiceConnection_SynthOutputFormat, value);
            }
        }
    }
}

File: source/bindings/java/com/microsoft/cognitiveservices/speech/dialog/BotConnectorConfig.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech.dialog;

import java.io.Closeable;

import com.microsoft.cognitiveservices.speech.SpeechConfig;
import com.microsoft.cognitiveservices.speech.util.Contracts;

/**
 * Class that defines configurations for speech bot connector.
 */
public class BotConnectorConfig extends SpeechConfig  {

    /*! \cond PROTECTED */
    static Class<?> botConnectorConfig = null;

    // load the native library.
    static {
        // trigger loading of native library
        try {
            Class.forName(SpeechConfig.class.getName());
        }
        catch (ClassNotFoundException ex) {
            throw new IllegalStateException(ex);
        }
        botConnectorConfig = BotConnectorConfig.class;
    }

    protected BotConnectorConfig(com.microsoft.cognitiveservices.speech.internal.BotConnectorConfig botConnectorConfigImpl) {
        super(botConnectorConfigImpl);
    }
    /*! \endcond */

    /**
     * Creates and instance of the bot connector config with the specified secret key.
     * @param secretKey Speech channel secret key.
     * @param subscription Service subscription key.
     * @param region Service region key.
     * @return The created config.
     */
    public static BotConnectorConfig fromSecretKey(String secretKey, String subscription, String region) {
        Contracts.throwIfNull(secretKey, "secretKey");
        Contracts.throwIfNull(subscription, "subscription");
        Contracts.throwIfNull(region, "region");
        return new BotConnectorConfig(com.microsoft.cognitiveservices.speech.internal.BotConnectorConfig.FromSecretKey(secretKey, subscription, region));
    }

    /**
     * Sets the Speech channel secret key..
     * @param value Speech channel secret key to set.
     */
    public void setSecretKey(String value) {
        Contracts.throwIfNullOrWhitespace(value, "value");
        getBotConfigImpl().SetSecretKey(value);
    }

    /**
     * Gets the Speech channel secret key.
     * @return Speech channel secret key.
     */
    public String getSecretKey() {
        return getBotConfigImpl().GetSecretKey();
    }

    /**
     * Sets the audio format that is returned by text to speech during a bot reply.
     * @param value The audio format to return.
     */
    public void setBotTextToSpeechAudioFormat(String value) {
        Contracts.throwIfNullOrWhitespace(value, "value");
        getBotConfigImpl().SetTextToSpeechAudioFormat(value);
    }

    /**
     * Gets the audio format that is returned by text to speech during a bot reply.
     * @return The audio format.
     */
    public void getBotTextToSpeechAudioFormat() {
        getBotConfigImpl().GetTextToSpeechAudioFormat();
    }

    /**
      * Returns the bot connector configuration.
      * @return The implementation of the config.
      */
    protected com.microsoft.cognitiveservices.speech.internal.BotConnectorConfig getBotConfigImpl() {
        return (com.microsoft.cognitiveservices.speech.internal.BotConnectorConfig) getImpl();
    }
}
